### PR TITLE
kernel-log-check: revise ignore_str for usb errors.

### PR DESCRIPTION
**Pull request by aiChaoSONG**
Fixes: #431 #514

### CODE REVIEW

File: tools/sof-kernel-log-check.sh [hunk 1]
```diff
@@ -164,8 +164,7 @@ ignore_str="$ignore_str"'|tpm tpm0: tpm_try_transmit: send\(\): error -5'
 ignore_str="$ignore_str"'|platform regulatory\.0: Direct firmware load for regulatory\.db failed with error -2'
 ignore_str="$ignore_str"'|cfg80211: failed to load regulatory\.db'
 ignore_str="$ignore_str"'|EXT4-fs \(nvme0n1p6\): re-mounted\. Opts: errors=remount-ro'
-ignore_str="$ignore_str"'|usb 2-3: Enable of device-initiated U1 failed\.'
-ignore_str="$ignore_str"'|usb 2-3: Enable of device-initiated U2 failed\.'
+ignore_str="$ignore_str"'|usb .-.+: (Enable|Disable) of device-initiated .+ failed\.'
```
Comment (xiulipan) [inline tools/sof-kernel-log-check.sh:167]:
```suggestion
ignore_str="$ignore_str"'|usb .+: (Enable|Disable) of device-initiated .+ failed\.'
```

**Review by xiulipan — CHANGES_REQUESTED** [review]
Interesting here, what is the `2-` means here? USB 2 or USB bus 2?
Do we want to ignore all USB 2 and 3 issues?

better with 

```
ignore_str="$ignore_str"'|hub [23]-.+:'
ignore_str="$ignore_str"'|usb [23]-.+:'
```
And remove all other duplicate filters

File: tools/sof-kernel-log-check.sh (hunk 1 above)
Comment (aiChaoSONG) [reply tools/sof-kernel-log-check.sh]: we can ignore error messages for usb hub, but not all usb. for hub, I use `hub [23]-.+'`, and for usb, I update to use specific message `usb [23]-.+: Failed to suspend device, error` from #431.

**Comment by marc-hb** [comment]
> Interesting here, what is the 2- means here? USB 2 or USB bus 2?

bus 2, check `lsusb -tv`

> ignore_str="$ignore_str"'|usb [23]+-.+:'

No this would ignore absolutely all USB errors, including from USB audio devices (e.g. #593) and from USB network devices that we use.

File: tools/sof-kernel-log-check.sh [hunk 2]
```diff
@@ -164,8 +164,7 @@ ignore_str="$ignore_str"'|tpm tpm0: tpm_try_transmit: send\(\): error -5'
 ignore_str="$ignore_str"'|platform regulatory\.0: Direct firmware load for regulatory\.db failed with error -2'
 ignore_str="$ignore_str"'|cfg80211: failed to load regulatory\.db'
 ignore_str="$ignore_str"'|EXT4-fs \(nvme0n1p6\): re-mounted\. Opts: errors=remount-ro'
-ignore_str="$ignore_str"'|usb 2-3: Enable of device-initiated U1 failed\.'
-ignore_str="$ignore_str"'|usb 2-3: Enable of device-initiated U2 failed\.'
+ignore_str="$ignore_str"'|usb .+: (Enable|Disable) of device-initiated .+ failed\.'
```
Comment (xiulipan) [inline tools/sof-kernel-log-check.sh:167]: I will do some clean up later to merge filter by model.

**Review by xiulipan — APPROVED** [review]
Then I think we filter all hub issues and some usb issues

**Comment by marc-hb** [comment]
Let's first stabilize the current test results with small, careful and focused changes. Once the filters have not changed a while then we can start from a "steady green base" and look at refactorings, cleanups and other simplifications.

Continuous Integration = small changes spread over time.

**Comment by xiulipan** [comment]
@marc-hb I see nothing with CI, are you good to merge this?

**Review by plbossart — DISMISSED** [review]

**Review by marc-hb — APPROVED** [review]

**Comment by marc-hb** [comment]
@aiChaoSONG there is git conflict now, probably because I just merged #624 sorry. Should be small.

**Comment by aiChaoSONG** [comment]
@marc-hb rebased, thanks.

File: tools/sof-kernel-log-check.sh [hunk 3]
```diff
@@ -189,8 +188,8 @@ ignore_str="$ignore_str"'|\[drm:drm_dp_send_link_address \[drm_kms_helper\]\] \*
 
 # CHT devices with USB hub, issues reported by sof-test
 # BugLink: https://github.com/thesofproject/sof-test/issues/431
-ignore_str="$ignore_str"'|hub 2-.: .'
-ignore_str="$ignore_str"'|usb 2-.: .'
+ignore_str="$ignore_str"'|hub [23]-.+'
+ignore_str="$ignore_str"'|usb [23]-.+: Failed to suspend device, error'
```
Comment (marc-hb) [inline tools/sof-kernel-log-check.sh:192]: I think the first line ignores everything that the second line does, so the second line is pointless.

The first line has two issues: 1. it ignores all USB errors on devices `2-9: ...`, it ignores zero error on devices like `2-10: ...`

> As we want to ignore all hub error but not all usb errors, ...

I don't think there is any concept of "hub error" vs "other errors", all errors are prefixed the same.

Can you please ignore only the errors reported in #431 ?

**Comment by aiChaoSONG** [comment]
The third-party issues is not seen even this is not merged, close.

**Review by keqiaozhang — APPROVED** [review]

**Pull request merged**